app/pages/household/[houseId]/index.tsx

```diff
@@ -43,6 +55,33 @@ const Login = () => {
     console.log("Document written with ID: ", docRef.id);
   };
 
+  const handleDeleteHouse = async (event: any) => {
+    event.preventDefault();
+
+    const docRef = await deleteDoc(doc(db, "household", householdRef.id));
+
+    if (modalCheckboxRef.current !== null) {
+      modalCheckboxRef.current.checked = !modalCheckboxRef.current.checked;
+    }
+    console.log("House deleted");
+  };
+
+  const handleLeaveHouse = async (event: any) => {
+    event.preventDefault();
+    const currentUserId = currentUser?.userId;
+    const docRef = doc(db, "household", householdRef.id);
+    try {
+      await updateDoc(docRef, { users: arrayRemove(currentUserId) });
+    } catch (error) {
+      console.log("error");
+      alert(error);
+    }
+    if (modalCheckboxRef.current !== null) {
+      modalCheckboxRef.current.checked = !modalCheckboxRef.current.checked;
+    }
+    console.log("House left");
+  };
+
```

Comment:
You can try use the useRouter hook to redirect user to the homepage after submitting a delete/leave house form, since right now its sits weird staying on the same page after pressing either button.
Check this link : https://stackoverflow.com/questions/70487236/redirect-to-another-page-based-on-form-input-in-next-js